com.unity.render-pipelines.universal/Runtime/2D/LightUtility.cs

```diff
@@ -170,127 +171,250 @@ public static Bounds GenerateSpriteMesh(Mesh mesh, Sprite sprite)
-            for (var i = 0; i < pointCount; ++i)
-                inputs[i] = new ContourVertex() { Position = new Vec3() { X = shapePath[i].x, Y = shapePath[i].y }};
-
-            var tessI = new Tess();
-            tessI.AddContour(inputs, ContourOrientation.Original);
-            tessI.Tessellate(WindingRule.EvenOdd, ElementType.Polygons, 3);
-
-            var indicesI = tessI.Elements.Select(i => i);
-            var verticesI = tessI.Vertices.Select(v => new float3(v.Position.X, v.Position.Y, 0));
-
-            var finalVertices = new NativeArray<ParametricLightMeshVertex>(verticesI.Count() + 2 * shapePath.Length, Allocator.Temp);
-            var finalIndices = new NativeArray<ushort>(indicesI.Count() + 6 * shapePath.Length, Allocator.Temp);
-
-            var vertexOffset = 0;
-            foreach(var v in verticesI)
+            float3 dvup = new float3(0, 1f, 0);
+            float3 dvrt = new float3(1f, 0, 0);
+
+            float dr = math.dot(dirNormalized, dvrt);
+            float du = math.dot(dirNormalized, dvup);
+            float cu = math.acos(du);
+            float sn = dr >= 0 ? 1.0f : -1.0f;
+            float an = cu * Mathf.Rad2Deg * sn;
+
+            // Adjust angles when direction is parallel to Up Axis.
+            an = (du != 1f) ? an : 0;
+            an = (du != -1f) ? an : -180f;
+            return (360.0f + an) % 360.0f;
+        }
+        
+        static void GetArc(NativeArray<float3> arcPoints, float3 from, float3 to, float3 pivot, float3 center, float radius, float detail)
+        {
+            var fq = SlopeAngle(math.normalize(from));
+            var tq = SlopeAngle(math.normalize(to));
+            var sq = SlopeAngle(math.normalize(pivot));
+
+            var fd = math.abs(sq - fq);
+            var td = math.abs(tq - sq);
+            var df = math.abs(fd - td);
+            
+            if (df > 1.0f)
+                fq = (fd > td) ? (sq + td) : (fq + 360);
+
+            var arcLength = tq - fq;
+            var ix = 0;
+            for (int i = 0; i <= detail; i++)
             {
-                finalVertices[vertexOffset++] = new ParametricLightMeshVertex
-                {
-                    position = v,
-                    color = meshInteriorColor
-                };
+                var x = Mathf.Sin(Mathf.Deg2Rad * fq) * radius;
+                var y = Mathf.Cos(Mathf.Deg2Rad * fq) * radius;
+                var pt = new float3(x, y, 0);
+                arcPoints[ix++] = (math.normalize(pt) * radius) + center;
+                fq += (arcLength / detail);
             }
+        }
 
-            var indexOffset = 0;
-            foreach (var v in indicesI)
+        static void GenerateBevels(NativeArray<float4> bevel, float fallOff, Color meshInteriorColor, NativeArray<ushort> indices,
+            NativeArray<ParametricLightMeshVertex> vertices, ref int vcount, ref int icount)
+        {
+            var vertexCount = bevel.Length;
+            var quality = 4;
```

Comment:
Should we make this configurable?

Reply:
Yes, let's discuss. @UnityPaul wanted minimal changes, so not yet exposed.

Reply:
The implementation has changed. Although we can choose the quality levels, let's keep default for now. Can change later.